src/javascript/app/pages/trade/contract.js

```diff
@@ -75,20 +75,24 @@ const Contract = (() => {
 
         const form_barrier = getFormNameBarrierCategory(form_name);
         form               = form_barrier.form_name;
-        barrier            = barrier_category = form_barrier.barrier_category;
+        if (!form) {
+            return;
+        }
+        barrier = barrier_category = form_barrier.barrier_category;
```

Comment:
Does anything break by removing the `barrier_category` and use `barrier` instead?

Reply:
no, I think in the past it was just used to show we are comparing category with category in the loop below.